synapse/config/workers.py

```diff
@@ -425,6 +368,54 @@ def _should_this_worker_perform_duty(
         # (By this point, these are either the same value or only one is not None.)
         return bool(new_option_should_run_here or legacy_option_should_run_here)
 
+    def _worker_names_performing_this_duty(
+        self,
+        config: Dict[str, Any],
+        legacy_option_name: str,
+        legacy_app_name: str,
+        modern_instance_list_name: str,
+    ) -> List[str]:
+        """
+        Retrieves the names of the workers handling a given duty, by either legacy
+        option or instance list
```

Comment:
The federation sender bit had some useful comments above it, I think:

```suggestion
        Retrieves the names of the workers handling a given duty, by either legacy
        option or instance list.

        There are two ways of configuring which instances handle a given duty, e.g.
        for configuring pushers:

        1. The old way where "start_pushers" is set to false and running a
          `synapse.app.pusher'` worker app.
        2. Specifying the workers sending federation in `pusher_instances`.
```